### PR TITLE
Merging robot relative drive into dev

### PR DESCRIPTION
**Pull request by BrandonBeaver117**
Adding robot relative drive suppor

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java
Comment (a-viv-a) [inline src/main/java/frc/robot/Constants.java]: Why are you setting offsets in a feature?

File: src/main/java/frc/robot/Constants.java
Comment (a-viv-a) [inline src/main/java/frc/robot/Constants.java]: Why are you setting offsets in a feature? This is fine, but it's better as a fix branch -- it can be merged independently of your feature, so it should be independent.

File: src/main/java/frc/robot/commands/RotateVectorDriveCommand.java
Comment (a-viv-a) [inline src/main/java/frc/robot/commands/RotateVectorDriveCommand.java:34]: a doccomment about what this member stores would be nice.

File: src/main/java/frc/robot/commands/RotateVectorDriveCommand.java
Comment (a-viv-a) [inline src/main/java/frc/robot/commands/RotateVectorDriveCommand.java]: this comment stayed in the same place, but the code it was referring to moved -- please dont do that

File: src/main/java/frc/robot/commands/RotateVectorDriveCommand.java
Comment (a-viv-a) [inline src/main/java/frc/robot/commands/RotateVectorDriveCommand.java]: ths boolean logic is reduntent and fairly ugly -- it could be expressed with a single if and a ternary.
`` if (Util.vectorMagnitude(rotX, rotY) > Drive.ROTATE_VECTOR_MAGNITUDE) ``
and then ``angle = isRobotRelative ? /** ect ect */``

File: src/main/java/frc/robot/commands/RotateVelocityDriveCommand.java
Comment (a-viv-a) [inline src/main/java/frc/robot/commands/RotateVelocityDriveCommand.java:42]: if this is just for testing, can we comment it out before we merge it? each shuffleboard value adds network traffic and cpu load.

**Review by a-viv-a — CHANGES_REQUESTED** [review]
needs a bit of work, but overall seems good -- nice work!

File: src/main/java/frc/robot/commands/RotateVectorDriveCommand.java
Comment (BrandonBeaver117) [reply src/main/java/frc/robot/commands/RotateVectorDriveCommand.java]: yeah that looks better

File: src/main/java/frc/robot/Constants.java
Comment (BrandonBeaver117) [reply src/main/java/frc/robot/Constants.java]: I wanted to test its driving, but modules needed calibration first. In hindsight, probably should have made a fix-branch first (like you said)

**Comment by a-viv-a** [comment]
@BrandonTheBusyBeaver117 this branch is out of date -- you need to merge `dev` -> `feat/robot-rel` for the module offset changes. Github provides a helpful button for it below.

**Review by a-viv-a — APPROVED** [review]
LGTM
feel free to merge if you want Brandon, or get a review from Lexie / Audrey

**Pull request merged**
